Fixes #29871 - Fix Hammer deprecation warning with lce

chris1984 opened the pull request:
Before on 6.5 below:
```bash
TASK [satellite-clone : Disassociate capsules with lifecycle environments (to avoid the cloned Satellite server talking with live capsules)] **********************************************************************
Friday 15 May 2020  12:35:40 -0400 (0:00:00.234)       1:07:09.580 ************ 
changed: [localhost]

stdout: All Capsules are unassociated with any lifecycle environments. This is to avoid any syncing errors with your original Satellite and any interference with existing infrastructure. To reverse these changes, run the following commands, making sure to replace the credentials with your own.
hammer -u admin -p changeme --csv capsule content add-lifecycle-environment --id 2 --environment-id 2

stderr: Warning: Option --environment-id is deprecated. Use --lifecycle-environment-id instead
```
After:

6.6:
```bash
TASK [satellite-clone : Disassociate capsules with lifecycle environments (to avoid the cloned Satellite server talking with live capsules)] *****************************************************************
Wednesday 20 May 2020  14:54:35 +0000 (0:00:00.076)       1:03:04.322 *********
changed: [localhost]

stdout: All Capsules are unassociated with any lifecycle environments. This is to avoid any syncing errors with your original Satellite and any interference with existing infrastructure. To reverse these changes, run the following commands, making sure to replace the credentials with your own.
hammer -u admin -p changeme --csv capsule content add-lifecycle-environment --id 2 --lifecycle-environment-id 2
```
6.5:
```bash
TASK [satellite-clone : Disassociate capsules with lifecycle environments (to avoid the cloned Satellite server talking with live capsules)] **********************************************************************
Monday 18 May 2020  16:00:31 -0400 (0:00:00.228)       1:08:06.776 ************ 
changed: [localhost]

stdout: All Capsules are unassociated with any lifecycle environments. This is to avoid any syncing errors with your original Satellite and any interference with existing infrastructure. To reverse these changes, run the following commands, making sure to replace the credentials with your own.
hammer -u admin -p changeme --csv capsule content add-lifecycle-environment --id 2 --environment-id 2
```

Starting to kick off tests on Dolly

chris1984 commented:
@johnpmitsch Dolly passed:

```bash
PLAY RECAP *********************************************************************
192.168.121.126            : ok=21   changed=16   unreachable=0    failed=0
192.168.121.139            : ok=21   changed=16   unreachable=0    failed=0
192.168.121.151            : ok=21   changed=16   unreachable=0    failed=0
192.168.121.31             : ok=21   changed=16   unreachable=0    failed=0

Wednesday 20 May 2020  16:38:35 -0400 (3:46:57.409)       3:57:19.826 *********
===============================================================================
test-clone : run satellite-clone playbook ---------------------------- 13617.41s
setup : install ansible ----------------------------------------------- 195.32s
setup : install the latest version of git ----------------------------- 150.57s
setup : register with subscription-manager ---------------------------- 108.64s
setup : install the latest version of wget ----------------------------- 83.77s
setup : enable RHEL 7 extras (for ansible) ----------------------------- 56.34s
setup : clone satellite-clone ------------------------------------------- 7.11s
setup : run reallocation script ----------------------------------------- 3.26s
Gathering Facts --------------------------------------------------------- 3.10s
setup : get backup files ------------------------------------------------ 2.68s
Gathering Facts --------------------------------------------------------- 1.58s
setup : copy satellite-clone-vars.yml ----------------------------------- 1.40s
setup : chmod 755 helper scripts ---------------------------------------- 1.33s
setup : rm satellite-clone dir (if exists) ------------------------------ 1.06s
setup : rebase PR branch ------------------------------------------------ 0.94s
setup : update backup_dir variable -------------------------------------- 0.85s
setup : check if satellite-clone-vars.yml exists ------------------------ 0.83s
setup : create backup directory ----------------------------------------- 0.82s
setup : checkout PR branch ---------------------------------------------- 0.81s
setup : stat ------------------------------------------------------------ 0.79s
```

johnpmitsch commented on roles/satellite-clone/files/disassociate_capsules.rb:
```diff
@@ -30,13 +30,19 @@ def get_info_from_hammer(command, column=1)
-      run_hammer_cmd("--csv capsule content remove-lifecycle-environment --id #{capsule[:id]} --environment-id #{env}")
-      reverse_command = prepare_hammer_cmd("--csv capsule content add-lifecycle-environment --id #{capsule[:id]} --environment-id #{env}")
-      reverse_commands << reverse_command
+      if satellite_version > 6.5
```
It shouldn't make a difference with two digits, but it is nice to use `Gem::Version` to compare versions to make sure the version math is done correctly. You'll need to add `require "rubygems"` at the top as well
```suggestion
      if Gem::Version.new(satellite_version) > Gem::Version.new(6.5)
```

chris1984 replied:
Fixed

johnpmitsch commented on roles/satellite-clone/files/disassociate_capsules.rb:
```diff
@@ -30,13 +30,19 @@ def get_info_from_hammer(command, column=1)
-      reverse_command = prepare_hammer_cmd("--csv capsule content add-lifecycle-environment --id #{capsule[:id]} --environment-id #{env}")
-      reverse_commands << reverse_command
+      if satellite_version > 6.5
+        run_hammer_cmd("--csv capsule content remove-lifecycle-environment --id #{capsule[:id]} --lifecycle-environment-id #{env}")
```
Can you not repeat `--csv capsule content remove-lifecycle-environment --id #{capsule[:id]} ` and `--csv capsule content add-lifecycle-environment --id #{capsule[:id]}`

You could define something like this:
```ruby
def capsule_lce_args(action, capsule_id, env, lifecycle=false)
  "--csv capsule content #{action}-lifecycle-environment --id #{capsule_id} --#{lifecycle ? "lifecycle-" : ""}environment-id #{env}"
end
```

and could use in the following formats:
```ruby
run_hammer_cmd(capsule_lce_args("remove", capsule[:id], env, true)) # 6.5+
prepare_hammer_cmd(capsule_lce_args("add", capsule[:id], env, true)) # 6.5+
run_hammer_cmd(capsule_lce_args("remove", capsule[:id], env, false)) # < 6.5
prepare_hammer_cmd(capsule_lce_args("add", capsule[:id], env, false)) # 6.5+
```

chris1984 replied:
Fixed

chris1984 commented:
Doing some manual testing with the new changes, then will kick off dolly

chris1984 commented:
@johnpmitsch Dolly passed with updated changes and manual tests are still good:

```shell
PLAY RECAP *********************************************************************
192.168.121.192            : ok=21   changed=16   unreachable=0    failed=0
192.168.121.22             : ok=21   changed=16   unreachable=0    failed=0
192.168.121.39             : ok=21   changed=16   unreachable=0    failed=0
192.168.121.91             : ok=21   changed=16   unreachable=0    failed=0

Wednesday 03 June 2020  15:08:54 -0400 (3:53:20.851)       4:11:57.110 ********
```
6.6:
```shell
TASK [satellite-clone : Disassociate capsules with lifecycle environments (to avoid the cloned Satellite server talking with live capsules)] *******************************************************************************************Wednesday 03 June 2020  11:09:35 -0400 (0:00:00.075)       1:12:16.294 ******** 
changed: [localhost] => {"changed": true, "rc": 0, "stderr": "", "stderr_lines": [], "stdout": "All Capsules are unassociated with any lifecycle environments. This is to avoid any syncing errors with your original Satellite and any interference with existing infrastructure. To reverse these changes, run the following commands, making sure to replace the credentials with your own.\nhammer -u admin -p changeme --csv capsule content add-lifecycle-environment --id 2 --lifecycle-environment-id 2\n", "stdout_lines": ["All Capsules are unassociated with any lifecycle environments. This is to avoid any syncing errors with your original Satellite and any interference with existing infrastructure. To reverse these changes, run the following commands, making sure to replace the credentials with your own.", "hammer -u admin -p changeme --csv capsule content add-lifecycle-environment --id 2 --lifecycle-environment-id 2"]}

stdout: All Capsules are unassociated with any lifecycle environments. This is to avoid any syncing errors with your original Satellite and any interference with existing infrastructure. To reverse these changes, run the following commands, making sure to replace the credentials with your own.
hammer -u admin -p changeme --csv capsule content add-lifecycle-environment --id 2 --lifecycle-environment-id 2
```
6.5:
```shell
TASK [satellite-clone : Disassociate capsules with lifecycle environments (to avoid the cloned Satellite server talking with live capsules)] **************************************************************Thursday 04 June 2020  18:24:48 -0400 (0:00:00.089)       1:06:43.992 ********* 
changed: [localhost] => {"changed": true, "rc": 0, "stderr": "", "stderr_lines": [], "stdout": "All Capsules are unassociated with any lifecycle environments. This is to avoid any syncing errors with your original Satellite and any interference with existing infrastructure. To reverse these changes, run the following commands, making sure to replace the credentials with your own.\nhammer -u admin -p changeme --csv capsule content add-lifecycle-environment --id 2 --environment-id 2\n", "stdout_lines": ["All Capsules are unassociated with any lifecycle environments. This is to avoid any syncing errors with your original Satellite and any interference with existing infrastructure. To reverse these changes, run the following commands, making sure to replace the credentials with your own.", "hammer -u admin -p changeme --csv capsule content add-lifecycle-environment --id 2 --environment-id 2"]}

stdout: All Capsules are unassociated with any lifecycle environments. This is to avoid any syncing errors with your original Satellite and any interference with existing infrastructure. To reverse these changes, run the following commands, making sure to replace the credentials with your own.
hammer -u admin -p changeme --csv capsule content add-lifecycle-environment --id 2 --environment-id 2
```
